### ISSUE TITLE
Impossibility of two class multilabel classification using GraphGym

### ISSUE THREAD

**Issue by RemyLau** [pyg-team/pytorch_geometric#7260]
### 🐛 Describe the bug

The current GraphGym model builder makes it impossible to set up multilabel tasks with **exactly** two classes. This is because of the implicit modification of `dim_out` in the model builder function.

https://github.com/pyg-team/pytorch_geometric/blob/8b37ad571b6e08d700f344cd9965724939f4bd4c/torch_geometric/graphgym/model_builder.py#L77-L78

I understand that in *some* cases, it is desirable to truncate the two-class binary task into a single binary classification task when it is ***actually*** a binary classification task. But there is currently no mechanism to ensure whether this is the case and *ALL* two-class classification tasks are treated as a single binary classification task.

I think it would be great to give the option at least to bypass this implicit modification. Or is there any way to work around this?

### Environment

* PyG version: latest
* PyTorch version: 2.0.0
* OS: Linux
* Python version: 3.10.9
* CUDA/cuDNN version: cuda 11.7
* How you installed PyTorch and PyG (`conda`, `pip`, source): `conda`
* Any other relevant information (*e.g.*, version of `torch-scatter`):

**Comment by rusty1s** [comment]
Do you mean that you cannot train a multi-label problem (both classes can be present at the same time) due to this? Wouldn't you want to set `task_type: classification_multi` in this case?

**Comment by RemyLau** [comment]
Hi @rusty1s, thanks for your reply! I did try to use `task_type: classification_multi`. But the issue persists because of the logic I linked earlier, which then turns `dim_out` to 1.

**Comment by rusty1s** [comment]
Oh, wow. I guess this should be "'classification' == cfg.dataset.task_type`. Does that resolve your issues?

**Comment by RemyLau** [comment]
Yes exactly. I think that should work
